### ISSUE TITLE
False-positive MismatchingDocblockParamType depending on union type order

### ISSUE THREAD

**Issue by fantomas662** [vimeo/psalm#2380]
`callable(T):string|int` produces `MismatchingDocblockParamType`, `callable(T):int|string` does not:

https://psalm.dev/r/35c70fac4a

**Comment by muglug** [comment]
This is _partly_ by design - an `int` can never be a valid `callable`, but a `string` can. On the other hand this should be a bug:

```php
/**
 * @param string $mapper
 */
function map2(callable $mapper): void {}

map2("foo");
```

**Comment by weirdan** [comment]
@fantomas662 are you looking to define callable returning `string|int`? If so you will need to wrap the return type in braces, like this: `callable(T):(string|int)`

**Comment by fantomas662** [comment]
@weirdan Yeah, that was it.

**Comment by muglug** [comment]
And now Psalm’s error should help you figure that out
